public/app/features/dimensions/editors/FileUploader.tsx

```diff
@@ -0,0 +1,132 @@
+import { css } from '@emotion/css';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import SVG from 'react-inlinesvg';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { FileDropzone, useStyles2, Button, DropzoneFile, Field } from '@grafana/ui';
+
+import { MediaType } from '../types';
+interface Props {
+  setFormData: Dispatch<SetStateAction<FormData>>;
+  mediaType: MediaType;
+  setUpload: Dispatch<SetStateAction<boolean>>;
+  newValue: string;
+  error: ErrorResponse;
+}
+interface ErrorResponse {
+  message: string;
+}
+export function FileDropzoneCustomChildren({ secondaryText = 'Drag and drop here or browse' }) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.iconWrapper}>
+      <small className={styles.small}>{secondaryText}</small>
+      <Button type="button" icon="upload">
+        Upload
+      </Button>
+    </div>
+  );
+}
+export const FileUploader = ({ mediaType, setFormData, setUpload, error }: Props) => {
+  const styles = useStyles2(getStyles);
+  const [dropped, setDropped] = useState<boolean>(false);
+  const [file, setFile] = useState<string>('');
+
+  const Preview = () => (
+    <Field label="Preview">
+      <div className={styles.iconPreview}>
+        {mediaType === MediaType.Icon && <SVG src={file} className={styles.img} />}
+        {mediaType === MediaType.Image && <img src={file} className={styles.img} />}
+      </div>
+    </Field>
+  );
+
+  const onFileRemove = (file: DropzoneFile) => {
+    fetch(`/api/storage/delete/upload/${file.file.name}`, {
+      method: 'DELETE',
+    }).catch((error) => console.error('cannot delete file', error));
+  };
+
+  const acceptableFiles =
+    mediaType === 'icon' ? 'image/svg+xml' : 'image/jpeg,image/png,image/gif,image/png, image/webp';
+  return (
+    <FileDropzone
+      readAs="readAsBinaryString"
```

Comment:
Not sure how this new storage system works, but would it be possible to use other formats here, in particular `readAsDataURL` for images?

Reply:
For our specific use case we use dataURL directly in css background images, so it's just easier. But in general I think base64 encoding is preferred for file transfers, although I could be wrong.